include/musica/micm/micm.hpp

```diff
@@ -109,22 +115,39 @@ namespace musica
         std::unique_ptr<micm::Rosenbrock>,
         std::unique_ptr<micm::RosenbrockStandard>,
         std::unique_ptr<micm::BackwardEuler>,
-        std::unique_ptr<micm::BackwardEulerStandard>>;
+        std::unique_ptr<micm::BackwardEulerStandard>
+        #ifdef MUSICA_ENABLE_CUDA
+          ,
+          std::unique_ptr<micm::CudaRosenbrock>,
+        #endif
+        >;
 
    public:
     SolverVariant solver_variant_;
 
     MICM(const Chemistry &chemistry, MICMSolver solver_type, int num_grid_cells);
     MICM() = default;
-    ~MICM() = default;
+    ~MICM() {
+      #ifdef MUSICA_ENABLE_CUDA
+      // Clean up CUDA resources
+      // This must happen before the MICM destructor completes because
+      // cuda must clean all of its runtime resources
+      // Otherwise, we risk the CudaRosenbrock destructor running after 
+      // the cuda runtime has closed
+      std::visit([](auto& solver) {
+        solver.reset();
+       }, solver_variant_);
+      micm::cuda::CudaStreamSingleton::GetInstance().CleanUp();
+      #endif
```

Comment:
I like this note and wish we could highlight it somehow in our knowledge sharing doc/talk/etc.